### PR TITLE
CM-1213 - update discussion in backend

### PR DESCRIPTION
**Pull request by watchtower314**
This PR has a corresponding frontend pr: https://github.com/daostack/common/pull/906

Ticket with demo: https://daostack1.atlassian.net/jira/software/projects/CM/boards/7?selectedIssue=CM-1213

### CODE REVIEW

File: packages/firebase/functions/src/discussion/business/updateDiscussion.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+export const updateDiscussion = async (payload: UpdateDiscussionPayload) : Promise<WriteResult> => {
+	
+  await validate<UpdateDiscussionPayload>(payload, updateDiscussionDataValidationSchema); 
+  
+  const {discussionId, messageOwner, userId} = payload;
+
+	if (messageOwner !== userId) {
+    throw new CommonError('You cannot update this discussion');
+	}
```
Comment (alexander2001i) [inline packages/firebase/functions/src/discussion/business/updateDiscussion.ts:33]: 🤮
```suggestion
if (messageOwner !== userId) {
    throw new CommonError('You cannot update this discussion');
}
```

File: packages/firebase/functions/src/discussion/database/updateLastMessage.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export const updateLastMessage = async (discussionId: string) : Promise<WriteResult> => {
+
+    const discussion = await discussionCollection
+      .doc(discussionId)
+      .update({
+        lastMessage: firestore.Timestamp.now(),
+      });
+
+    return discussion
+}
```
Comment (alexander2001i) [inline packages/firebase/functions/src/discussion/database/updateLastMessage.ts:14]: Nope. That is not the way to go. The way to go is to have only updateMessage in the database methods and updateLastMessage in the business

**Review by alexander2001i — CHANGES_REQUESTED** [review]
Please, please, please fix the indentation 😄

File: packages/firebase/functions/src/discussion/business/updateDiscussion.ts (hunk 1 above)
Comment (watchtower314) [reply packages/firebase/functions/src/discussion/business/updateDiscussion.ts:33]: @alexander2001i  Sorryyyy, I'll add the line break, but this is how it looks for me, isn't it ok?
Screen Shot 2021-02-04 at 3 51 01

File: packages/firebase/functions/src/discussion/database/updateLastMessage.ts (hunk 2 above)
Comment (watchtower314) [reply packages/firebase/functions/src/discussion/database/updateLastMessage.ts:14]: Although we're not updating a message, we're updating the `lastMessage` field in a discussion, 
so I figured have `updateDiscussion` in business (maybe the name for this is wrong, though, and should also be something like updateLastMessage ) and  `updateLastMessage` in database 🤔
what do you think?

File: packages/firebase/functions/src/discussion/business/updateDiscussion.ts (hunk 1 above)
Comment (alexander2001i) [reply packages/firebase/functions/src/discussion/business/updateDiscussion.ts:33]: @watchtower314 Hhmmmm, it is ok, mb it was displayed incorrectly for me? Sorry

**Review by alexander2001i — APPROVED** [review]

**Review by lpetkov-sw — APPROVED** [review]

**Comment by lpetkov-sw** [comment]
Like the clean code followed by the Alex's rules!

**Pull request merged**